src/test/java/com/teragrep/cfe_39/CombinedFullTest.java

```diff
@@ -0,0 +1,309 @@
+        // MapReduce functionalities of the Hadoop cluster: https://hadoop.apache.org/docs/stable/hadoop-mapreduce-client/hadoop-mapreduce-client-core/MapReduceTutorial.html
+        // Avro side of documentations for MapReduce: https://avro.apache.org/docs/1.11.1/mapreduce-guide/
+
+        // Another method for pruning aside using avro-mapred is to use modification timestamp of the avro-file stored in HDFS:
+        // fs.setTimes(new Path(path+"/"+0.8), Long.parseUnsignedLong("1675930598000"), -1);
+        // where mtime is modification time and atime is access time. -1 as input parameter leaves the original atime/mtime value as is.
+        FileStatus[] fileStatuses = fs.listStatus(new Path(newDirectoryPath + "/"));
+        long count = Arrays.stream(fileStatuses).count();
+        if (count != 0) {
+            if (config.getPruneOffset() != 157784760000L) {
+                Assertions.fail("There are files available in the database when there should be none.");
+            }
+            boolean delete = false;
+            for (FileStatus a : fileStatuses) {
+                // Delete old files
+                if (a.getModificationTime() < System.currentTimeMillis()) {
+                    delete = fs.delete(a.getPath(), true);
+                    Assertions.assertTrue(delete);
+                    LOGGER.info("Deleted file {}", a.getPath());
+                }
+            }
+            Assertions.assertTrue(delete);
+            LOGGER.info("All files were pruned properly.");
+        }else {
+            if (config.getPruneOffset() == 157784760000L) {
+                Assertions.fail("There were no files available in the database when there should be.");
+            }
+            LOGGER.info("No files available as they were pruned properly already!");
+        }
+
+        fs.close();
+    }
+
+    public void hdfsReadCheck() throws IOException {
+        // Check that the files were properly written to HDFS with a read test.
+        String hdfsuri = config.getHdfsuri();
+
+        String path = config.getHdfsPath()+"/"+"testConsumerTopic";
+        // ====== Init HDFS File System Object
+        Configuration conf = new Configuration();
+        // Set FileSystem URI
+        conf.set("fs.defaultFS", hdfsuri);
+        // Because of Maven
+        conf.set("fs.hdfs.impl", org.apache.hadoop.hdfs.DistributedFileSystem.class.getName());
+        conf.set("fs.file.impl", org.apache.hadoop.fs.LocalFileSystem.class.getName());
+        // Set HADOOP user
+        System.setProperty("HADOOP_USER_NAME", "hdfs");
+        System.setProperty("hadoop.home.dir", "/");
+        //Get the filesystem - HDFS
+        FileSystem fs = FileSystem.get(URI.create(hdfsuri), conf);
+
+        //==== Create directory if not exists
+        Path workingDir=fs.getWorkingDirectory();
+        Path newDirectoryPath= new Path(path);
+        if(!fs.exists(newDirectoryPath)) {
```

Comment:
Same as before, conditionals should not be part of tests

Reply:
Solved from commit 65342c2e865912f981dfe82f470cab6884f72cdb onward